src/contact.rs

```diff
@@ -1094,4 +1096,104 @@ mod tests {
             vec![("Name one", "Address one"), ("Name two", "Address two")]
         )
     }
+
+    #[test]
+    fn test_get_contacts() {
+        let context = dummy_context();
+        let contacts = Contact::get_all(&context.ctx, 0, Some("some2")).unwrap();
+        assert_eq!(contacts.len(), 0);
+
+        let id = Contact::create(&context.ctx, "bob", "bob@mail.de").unwrap();
+        assert_ne!(id, 0);
+
+        let contacts = Contact::get_all(&context.ctx, 0, Some("bob")).unwrap();
+        assert_eq!(contacts.len(), 1);
+
+        let contacts = Contact::get_all(&context.ctx, 0, Some("alice")).unwrap();
+        assert_eq!(contacts.len(), 0);
+    }
+
+    #[test]
+    fn test_add_or_lookup() {
```

Comment:
this test became a bit long and not very unit-like.  due to rusts poor test framework we've often done this admittedly, but maybe having to scroll for a test is too much?  i don't know

Reply:
hm, breaking the tests into smaller pieces would probably result in even more code - some of the things checked in the last lines need more or less complicated data to be created first. 

of course, helper-function would help, but i am not sure if that is worth the effort just now.

but i think it is pretty straight-forward and easy enough to understand.